audit_engine/dynamic_analysis/adversarial_fuzz.py

```
@@ -0,0 +1,171 @@
+        """
+        This remains as a fallback for any custom output parsing you want to add if needed.
+        """
+        try:
+            raw = json.loads(output)
+            findings: List[Finding] = []
+            for item in raw:
+                finding = Finding(**item)
+                findings.append(finding)
+            return findings
+        except Exception as e:
+            raise ToolError(f"Failed to parse output: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)